src/main/java/ratpack/zipkin/ServerTracingModule.java

```diff
@@ -93,26 +101,29 @@ public ClientResponseInterceptor clientResponseInterceptor(final Brave brave) {
   }
 
   @Provides
-  public Brave getBrave(final Config config, final ServerConfig serverConfig) {
-    Brave.Builder braveBuilder = new Brave.Builder(
-        new RatpackServerClientLocalSpanState(
-                buildLocalEndpoint(
-                        config.serviceName,
-                        serverConfig.getPort(),
-                        serverConfig.getAddress()
-                )
-        )
-    );
+  public Brave getBrave(final Tracer tracer) {
+    return TracerAdapter.newBrave(tracer);
```

Comment:
Can you use `public static Brave newBrave(Tracer tracer, ServerClientAndLocalSpanState state)` instead?  That would allow you to use `RatpackServerClientLocalSpanState` as the instead of the thread local one.